src/array.cr

```diff
@@ -1770,4 +1770,12 @@ class Array(T)
       end
     end
   end
+
+  def bsearch
+    idx=(0...self.size).bsearch{|i|yield self[i]}
+    idx ? self[idx] : nil
```

Comment:
Order of method definition doesn't matter. By the time the `bsearch` method is invoked, `bsearch_index` will be defined.